xdf2fieldtrip.m

```diff
@@ -65,6 +64,43 @@
   end
 end
 
+% Read the EEG stream to get the first time stamp and sampling rate
+EEG_info=[];
+for i=1:length(streams)
+    if (strcmp(streams{i}.info.type,'EEG'))
+        EEG_t1= str2double(streams{i}.info.first_timestamp);
+        EEG_srate=streams{i}.info.effective_srate;
+        EEG_info=[EEG_t1 EEG_srate];
+    end
+end
+
+if isempty(EEG_info)
+  ft_error('no EEG streams were selected');
+end
+
+% Read events from the non continuous Marker stream
+event = [];
+for i=1:length(streams)
+    if (strcmp(streams{i}.info.type,'Markers'))
+        try
+            events = struct('sample', [], 'offset', [], 'duration', num2cell(ones(1, length(streams{i}.time_stamps))),...
+                'type', cellstr(repmat({'Marker'}, 1, length(streams{i}.time_stamps))), 'value', ' ', 'timestamp', []); 
+            for k=1:length(streams{i}.time_stamps)
+                if iscell(streams{i}.time_series)
+                    events(k).value = streams{i}.time_series{k};
+                else
+                    events(k).value = num2str(streams{i}.time_series(k));
+                end
+                events(k).sample = round((streams{i}.time_stamps(k)- EEG_t1)*EEG_srate);
```

Comment:
event samples are expressed relative to the EEG sampling rate. What if there are two EEG streams with different sampling rates? Or what if another data stream (e.g. audio or high speed eyetracker) has a higher sampling rate?  The resampling of all streams further down uses the highest sampling rate, not per see the EEG sampling rate. 

samples as specified in the event structure should match that of the resampled+appended data.